File: models/product.model.js
const { model, Schema } = require('mongoose');

const productSchema = new Schema({
    title: {
        type: String,
        required: true,
    },
    userId: {
        type: String,
        required: true,
    },
    phone: {
        type: String,
        required: true,
    },
    description: {
        type: String
    },
    price: {
        type: String
    },
    brand:{
        type: String
    },
    imageURL:{
        type: String
    }
});

const product = model('Product', productSchema);

module.exports = product;

File: lib/models/productmodel.dart
class ProductModel {
  String? title;
  String? userId;
  String? phone;
  String? description;
  String? price;
  String? brand;
  String? imageURL;

  ProductModel({
    this.title,
    this.userId,
    this.phone,
    this.description,
    this.price,
    this.brand,
    this.imageURL,
  });

  ProductModel.fromJson(Map<String, dynamic> json) {
    title = json['title'];
    userId = json['userId'];
    phone = json['phone'];
    description = json['description'];
    price = json['price'];
    brand = json['brand'];
    imageURL = json['imageURL'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = {};
    data['title'] = title;
    data['userId'] = userId;
    data['phone'] = phone;
    data['description'] = description;
    data['price'] = price;
    data['brand'] = brand;
    data['imageURL'] = imageURL;
    return data;
  }
}
